src/main/webapp/dataverse.xhtml

```diff
@@ -330,8 +331,9 @@
                                                                             </div>
                                                                         </td>
                                                                         <td>
-                                                                            <p:selectOneRadio id="OptionsRadio#{dsft.id}" value="#{dsft.requiredDV}" widgetVar="OptButton#{dsft.id}" disabled ="#{DataversePage.editInputLevel == false}"
-                                                                                              rendered="#{!dsft.hasChildren and !((!dsft.hasChildren  and dsft.required) or (dsft.hasChildren and dsft.hasRequiredChildren))}">
+                                                                            <p:selectOneRadio id="OptionsRadio#{dsft.id}" value="#{dsft.requiredDV}" widgetVar="OptButton#{dsft.id}" disabled ="#{DataversePage.editInputLevel == false}" 
+                                                                                              onclick="#{DataversePage.toggleInputLevel(mdb.id, dsft.id)}"
```

Comment:
what is this onclick doing? onclick is for calling javascript, but what you have here is a backing bean (java) call, so I don't think this would be being called when you click?

Reply:
I put it in because in my pre-testing I thought I had seen a case where changes to required/optional weren't being saved and because the method in the backing bean wasn't being used anywhere. I thought they were related so I put in the onclick. After removing both it seems to work so I'll chalk it up to bad pre-testing.